lib/pltr/v2/selectors/project.js

```diff
@@ -29,3 +37,61 @@ export const fileURLSelector = (state) => state.project.fileURL
 export const isDeviceFileSelector = createSelector(fileURLSelector, (fileURL) =>
   isDeviceFileURL(fileURL)
 )
+const emptyFileState = emptyFile('DummyFile', '2022.11.2')
+export const hasAllKeysSelector = (state) => {
+  const withoutSystemKeys = difference(Object.keys(state), SYSTEM_REDUCER_KEYS)
+  return difference(Object.keys(emptyFileState), withoutSystemKeys).length === 0
+}
+export const canSaveSelector = createSelector(
+  fileURLSelector,
+  isResumingSelector,
+  isOfflineSelector,
+  offlineModeEnabledSelector,
+  hasAllKeysSelector,
+  isCloudFileSelector,
+  (fileURL, isResuming, isOffline, offlineModeEnabled, hasAllKeys, isCloudFile) => {
+    return (
+      !!fileURL &&
+      !isResuming &&
+      (!isCloudFile || !(isCloudFile && isOffline && !offlineModeEnabled)) &&
```

Comment:
Hmmm, it does look complicated.

Here's my reasoning:
The entire expression evaluates to `true` if we *can* save (where "save" means save to disk because the concept doesn't exist on Firebase.)
Generally, you can't save a cloud file, but you are allowed to when it's a cloud file and `offlineMode` is enabled.
Later, in `save.js`, we check whether we should save an offline file.
If we're allowed to save, *but* we can't save an offline file, and we're a pro file, then `save` resolves *without* saving.

I might've made a mistake here, so let's apply DeMorgan's theorem:
```
!isCloudFile || !(isCloudFile && isOffline && !offlineModeEnabled)
= !isCloudFile || (!isCloudFile || !isOffline || offlineModeEnabled)
= !isCloudFile || !isCloudFile || !isOffline || offlineModeEnabled
= !isCloudFile || !isOffline || offlineModeEnabled
```
So it ends up being either:
 - we're not a cloud file, or
 - we're not offline, or
 - offline mode is enabled.

I think the offline mode check *might* be redundant here, but I like double-checking it in case the logic for checking whether to save an offline file instead of an ordinary file changes.

Does it make sense to you?
Should we apply that simplification I made to reason about the logic?